src/thor/timedep_forward.cc

```diff
@@ -149,17 +148,40 @@ inline bool TimeDepForward::ExpandForwardInner(GraphReader& graphreader,
                                                const valhalla::Location& destination,
                                                std::pair<int32_t, float>& best_path) {
 
-  // Skip shortcut edges for time dependent routes. Also skip this edge if permanently labeled
-  // (best path already found to this directed edge), if no access is allowed to this edge
+  // Skip shortcut edges for time dependent routes, if no access is allowed to this edge
   // (based on costing method), or if a complex restriction exists.
   bool has_time_restrictions = false;
-  if (meta.edge->is_shortcut() || meta.edge_status->set() == EdgeSet::kPermanent ||
-      !costing_->Allowed(meta.edge, pred, tile, meta.edge_id, localtime, nodeinfo->timezone(),
-                         has_time_restrictions) ||
-      costing_->Restricted(meta.edge, pred, edgelabels_, tile, meta.edge_id, true, localtime,
-                           nodeinfo->timezone())) {
+  if (meta.edge->is_shortcut() || !costing_->Allowed(meta.edge, pred, tile, meta.edge_id, localtime,
+                                                     nodeinfo->timezone(), has_time_restrictions)) {
     return false;
   }
+  bool is_complex_restriction = costing_->Restricted(meta.edge, pred, edgelabels_, tile, meta.edge_id,
```

Comment:
i think we can also have multi edge restrictions without a time component in bidirectional A*. i dont see why a multi edge restriction would always have to have a time component cc @gknisely

Reply:
sorry didnt see this already mentioned in the PR description